ui/plugin-system/src/model/panels.ts

```diff
@@ -24,14 +24,15 @@ export type PanelOptionsEditorComponent<T> = Pick<OptionsEditorTab, 'label'> & {
  */
 export interface PanelPlugin<Spec = UnknownSpec> extends Plugin<Spec> {
   PanelComponent: React.ComponentType<PanelProps<Spec>>;
-  /**
-   * React component for the content of "Query" tab
-   */
-  PanelQueryEditorComponent?: React.ComponentType<QueryEditorProps<Spec>>;
   /**
    * React components for custom tabs
    */
   panelOptionsEditorComponents?: Array<PanelOptionsEditorComponent<Spec>>;
+  /**
+   * If true, query editor will be hidden for panel plugin
+   * @default false
+   */
+  hideQueryEditor?: boolean;
```

Comment:
Nice. I suppose this is more explicit than `supportedQueryInterfaces`

Reply:
Something like `supportedQueryInterfaces` would be potentially more future-proof for if we support additional types of queries. 🤔 

Although I think this is fine for the sake of Eunice being able to move the refactor forward. We can cross that bridge when we get to it, especially knowing it could be pretty far away, since we're pretty focused on time series data from prometheus.